core/sys/energest.h

```diff
@@ -105,6 +105,21 @@ extern energest_t energest_leveldevice_current_leveltime[ENERGEST_CONF_LEVELDEVI
 										energest_current_time[type]); \
 										energest_current_mode[type] = 0; \
                                        } while(0)
+
+#define ENERGEST_SWITCH(type_off, type_on) do { \
+                                             rtimer_clock_t now = RTIMER_NOW(); \
```

Comment:
Well actually it can not. The life time of a variable is limited to a block. Also block-local variables hide any variable outside the block. Thus the only risks are confusing compile warnings or debug output.